Sources/IdentityState.swift

```diff
@@ -42,6 +42,9 @@ class IdentityState {
             if let ecid = identityProperties.getEcidFromDirectIdentityPersistence() {
                 identityProperties.ecid = ecid.ecidString // get ECID from direct extension
                 Log.debug(label: IdentityConstants.LOG_TAG, "IdentityState - Loading ECID from direct Identity extension on bootup '\(ecid)'")
+            } else if isIdentityDirectRegistered(event: event, getSharedState: getSharedState) {
+                Log.debug(label: IdentityConstants.LOG_TAG, "IdentityState - bootup direct Identity extention is registered, waiting for its state change.")
+                return false // If no ECID to migrate but Identity direct is registered, wait for Identity direct shared state
```

Comment:
I'm not handling the case where Identity direct is registered but the shared state doesn't contain an ECID (opted-out case). I'll need to update this.